mediaflow_proxy/utils/http_utils.py

```
@@ -1,644 +1,748 @@
-
-
-async def download_file_with_retry(url: str, headers: dict):
+    try:
+        logger.info(f"Extracting Sportsonline stream data from: {destination}")
+        extractor = ExtractorFactory.get_extractor("Sportsonline", proxy_headers.request)
+        result = await extractor.extract(destination)
+        logger.info(f"Sportsonline extraction successful. Stream URL: {result.get('destination_url')}")
+        _sportsonline_extraction_cache[destination] = {"data": result, "timestamp": current_time}
+        logger.info(f"Sportsonline data cached for {_sportsonline_cache_duration}s")
+        return result
+    except (ExtractorError, DownloadError, Exception) as e:
```

Ruff (0.14.6)

Do not catch blind exception: `Exception`

(BLE001)